src/Functions/FunctionsLogical.h

```diff
@@ -148,14 +154,23 @@ class FunctionAnyArityLogical : public IFunction
     }
 
     bool isVariadic() const override { return true; }
+    bool isShortCircuit(ShortCircuitSettings * settings, size_t /*number_of_arguments*/) const override
```

Comment:
It is strange why do we use `ShortCircuitSettings *`.
Maybe, use `ShortCircuitSettings &`, or return this struct from function.